### PR TITLE
feat: add graphql server

### PR DESCRIPTION
**Pull request by pedrobarco**

### CODE REVIEW

**Review by cbartolomeu — APPROVED** [review]

File: examples/quotes-api/cmd/quotes-api/main.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ func main() {
 	}
 	defer cleanup()
 
-	api.server.GET("/", api.quoteController.GetQuote)
+	// api.server.GET("/", api.quoteController.GetQuote)
```
Comment (Bernard0-dev) [inline examples/quotes-api/cmd/quotes-api/main.go]: cant we just remove this?

File: pkg/server/provider.go [hunk 2]
```diff
@@ -27,4 +27,8 @@ var (
 		ProvideTracerProvider,
 		NewServer,
 	)
+	GraphQLServerProviderSet = wire.NewSet(
```
Comment (Bernard0-dev) [inline pkg/server/provider.go:30]: if this is `GraphQLServerProviderSet`, shouldn't `ServerProviderSet` be something like `HTTPServerProviderSet` or `RestServerProviderSet`?

File: examples/quotes-api/cmd/quotes-api/wire.go [hunk 3]
```diff
@@ -47,8 +48,9 @@ func initializeAPI() (*API, func(), error) {
 		log.LoggerProviderSet,
 		yeapi.ClientProviderSet,
 		quote.GetQuoteUseCaseProviderSet,
+		graph.ResolverProviderSet,
 		rest.QuoteControllerProviderSet,
-		server.ServerProviderSet,
+		server.GraphQLServerProviderSet,
 		wire.Value(server.Version("1.0.0")),
 		wire.Value([]httpds.Plugin(nil)),
 		ProvideHealthChecks,
```
Comment (Bernard0-dev) [inline examples/quotes-api/cmd/quotes-api/wire.go]: why does the example only use `GraphQLServerProviderSet`?
shouldn't we have 2 examples? one for GQL and one for rest?

File: examples/quotes-api/internal/http/graph/schema.resolvers.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return []*model.Quote{
+		{
+			ID:   uuid.NewString(),
+			Text: q.Value,
+		},
+	}, nil
```
Comment (Bernard0-dev) [inline examples/quotes-api/internal/http/graph/schema.resolvers.go:27]: this part is not automatically generated right?

File: pkg/server/provider.go (hunk 2 above)
Comment (pedrobarco) [reply pkg/server/provider.go:30]: Yap, we should rename the REST one to `RESTServerProviderSet`

File: examples/quotes-api/cmd/quotes-api/wire.go (hunk 3 above)
Comment (pedrobarco) [reply examples/quotes-api/cmd/quotes-api/wire.go]: Sure, I'll clone this one and add it to `cmd/`

File: examples/quotes-api/internal/http/graph/schema.resolvers.go (hunk 4 above)
Comment (pedrobarco) [reply examples/quotes-api/internal/http/graph/schema.resolvers.go:27]: Correct, this is the actual resolver implementation.

File: examples/quotes-api/cmd/quotes-api/main.go (hunk 1 above)
Comment (pedrobarco) [reply examples/quotes-api/cmd/quotes-api/main.go]: I've split quotes-api `/cmd` folder into two different APIs:
```sh
examples/quotes-api/
├── cmd
│   ├── quotes-graphql-api
│   │   ├── config.go
│   │   ├── main.go
│   │   ├── wire.go
│   │   └── wire_gen.go
│   └── quotes-rest-api
│       ├── config.go
│       ├── main.go
│       ├── wire.go
│       └── wire_gen.go
```

**Review by Bernard0-dev — APPROVED** [review]

**Pull request merged**